mitodl/open-discussions
# Added title and channel_type to Channel to avoid many reddit requests

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #1658 

#### What's this PR do?
Adds `channel_type` and `title` to `Channel` so that we can avoid N+1 trips to reddit triggered by accessing `Submission.subreddit`, since the frontpage and channel listing APIs reddit provides don't provide the channel title or type

#### How should this be manually tested?

- Ensure you can still load the frontpage and channel pages
- Verify you have channels with `None` set on these new fields:
```python
from channels.models import *
from django.db.models import Q
Channel.objects.filter(Q(title__isnull=True) | Q(channel_type__isnull=True) | Q(allowed_post_types__isnull=True)).count()
```
- Run `./manage.py backpopulate_channel_fields`
- Run the check script again and verify no channel match
- Retest the frontpage and channel pages and verify they work, you should see a noticeable performance improvement even locally
